v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -0,0 +1,647 @@
+import assert from "node:assert/strict";
+import { before, describe, it } from "node:test";
+
+import { createHardhatRuntimeEnvironment } from "@nomicfoundation/hardhat-core";
+import { ParameterType, task } from "@nomicfoundation/hardhat-core/config";
+import { HardhatRuntimeEnvironment } from "@nomicfoundation/hardhat-core/types/hre";
+import {
+  NewTaskDefinition,
+  Task,
+} from "@nomicfoundation/hardhat-core/types/tasks";
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+
+import { parseTaskAndArguments } from "../../../src/internal/cli/main.js";
+
+interface TaskRes {
+  task: Task;
+  taskArguments: Record<string, any>;
+}
+
+describe("main", function () {
+  let hre: HardhatRuntimeEnvironment;
+
+  // TASKS and SUBTASKS are defined in the "before()" hooks before every "functionality test groups".
+  let TASK: NewTaskDefinition;
```

Comment:
I think I'd keep these as camelCase as they're not constants.

Reply:
[Done in  this commit](https://github.com/NomicFoundation/hardhat/pull/5282/commits/75bcb877e7701e0a746b8d147c18693414309f68)